lib/iris/tests/unit/fileformats/netcdf/saver/test_Saver__ugrid.py

```diff
@@ -1215,63 +1258,19 @@ def _check_two_different_meshes(self, vars):
             )
 
         # mesh2
-        self.assertEqual(vars["node_x_0"][_VAR_DIMS], ["Mesh2d_nodes_0"])
+        self.assertEqual(vars["node_x_0"][_VAR_DIMS], ["Mesh2d_nodes"])
         self.assertEqual(vars["face_x_0"][_VAR_DIMS], ["Mesh2d_faces_0"])
         self.assertEqual(
             vars["mesh2d_faces_0"][_VAR_DIMS],
             ["Mesh2d_faces_0", "Mesh2d_0_face_N_nodes"],
         )
         if "edge_coordinates" in vars["Mesh2d_0"]:
-            self.assertEqual(vars["longitude_0"][_VAR_DIMS], ["Mesh2d_edge_0"])
+            self.assertEqual(vars["longitude_0"][_VAR_DIMS], ["Mesh2d_edge"])
             self.assertEqual(
                 vars["mesh2d_edge_0"][_VAR_DIMS],
-                ["Mesh2d_edge_0", "Mesh2d_0_edge_N_nodes"],
+                ["Mesh2d_edge", "Mesh2d_0_edge_N_nodes"],
             )
 
-    def test_multiple_equal_mesh(self):
```

Comment:
NOTE: Actually not much has _really_ changed here  : see https://github.com/SciTools/iris/pull/6004/files#r1638046009